magma/backend/coreir_transformer.py

```diff
@@ -207,7 +207,7 @@ def connect(self, module_defn, port, value, non_input_ports):
         def get_source():
             if isinstance(value, Wireable):
                 return value
-            if isinstance(value, Array) and is_const(value):
+            if isinstance(value, Array) and is_const_digital(value):
```

Comment:
```suggestion
            if isinstance(value, Array) and issubclass(type(value).T, Digital) and value.const():
```

Reply:
we can just capture all that logic inline here. Then we can get rid of `is_const()` and the import of it. 

`all(x.const() for x in array)` duplicates `Array.const()` implementation.

Reply:
Actually we should probably do 

```python
if isinstance(value, Bits) and value.const()
```

because either (a) if you do `Array[N, Bit]` then you should really just do `Bits[N]`, or (b) if you do `Array[N, Clock](0)` or something then that should be different in the backend than a const bits (b/c clock types are differentiated). But not sure if external code is doing (a) or (b) and expecting it to work.

Reply:
don't know if const Clock exists and doubt it's being used (b)
with this change if you do Array[N, Bit](0) you'll get an array of Bit(0) consts, which doesn't seem unreasonable, but one could argue that the old logic generates more readable code.

Reply:
Yeah. I'm ok with either

`if isinstance(value, Bits) and value.const():`

or 

`if isinstance(value, Array) and issubclass(type(value).T, Bit) and value.const():`